cmd/examples/login-user-registry/server/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	cmd_utils "github.com/jinfwhuang/ds-toolkit/go-pkg/cmd-utils"
+	eth_client "github.com/jinfwhuang/ds-toolkit/go-pkg/ds"
+	protoId "github.com/jinfwhuang/ds-toolkit/proto/identity"
+)
+
+var (
+	ethAddr = &cli.StringFlag{
+		Name:  "eth-address",
+		Usage: "TODO: xxx",
+		Value: "http://127.0.0.1:8545",
+	}
+
+	contractAddr = &cli.StringFlag{
+		Name:  "contract-address",
+		Usage: "TODO: xxx",
+		Value: "0xC26f4289DFB6138C5f12e7D52D283F0Ce15FF985",
+	}
+
+	appFlags = []cli.Flag{
+		cmd_utils.GrpcPort,
+		cmd_utils.LogLevel,
+		cmd_utils.LogCaller,
+		ethAddr,
+		contractAddr,
+	}
+
+	callOpt = &bind.CallOpts{
+		Pending: true,
+	}
+)
+
+func main() {
+	app := cli.App{}
+	app.Name = "UserRegistryLogin"
+	app.Flags = appFlags
+	app.Action = start
+	if err := app.Run(os.Args); err != nil {
+		logrus.Info(err)
+	}
+}
+
+func start(cliCtx *cli.Context) error {
+	cmd_utils.SetupLogrus(cliCtx)
+
+	stop := make(chan struct{})
+
+	// Tcp Listener
+	grpcPort := cliCtx.Int(cmd_utils.GrpcPort.Name)
+	address := fmt.Sprintf("%s:%d", "0.0.0.0", grpcPort)
+	logrus.Infof("grpc address: %s", address)
+	lis, err := net.Listen("tcp", address)
+	if err != nil {
+		logrus.Fatalf("could not listen to port in Start() %s: %v", address, err)
+	}
+	// GRPC
+	grpcServer := grpc.NewServer()
+	userRegistryServer := NewUserRegistryServer(ethAddr.GetValue(), contractAddr.GetValue())
```

Comment:
Have you tried `ethAddr.GetValue()`, does it retrieve the value from command line argument? 

If not, try `cliCtx.String(ethAddr.Name)` instead?

Reply:
Both seem to work in this case. However, when I tried assigning a flag value in another part of the code, it did not work when I did ```ethAddr.GetValue()``` so I changed it to the other option.